File: TestWebApp/Bezahlwebsite/ClientApp/src/Interfaces/Payment.ts
import { ApplicationUser } from "./ApplicationUser";
import { BoughtArticle } from "./BoughtArticle";

export interface Payment {
  paymentId: number;
  personId: string;
  person: ApplicationUser;
  date: string;
  description: string;
  cashAmount: number;
  articles: BoughtArticle[];
  executorId: string | null;
  executor: ApplicationUser | null;
}

File: TestWebApp/DataAccess/Model/Payment.cs
namespace DataAccess.Models
{
    public class Payment
    {
        public int PaymentId { get; set; }
        public string PersonId { get; set; }
        public ApplicationUser Person { get; set; }
        public DateTime Date { get; set; }
        public string Description { get; set; }
        public double CashAmount { get; set; }
        public virtual List<BoughtArticle> Articles { get; set; } = new List<BoughtArticle>();        
        public string? ExecutorId { get; set; }
        public ApplicationUser? Executor { get; set; }
    }
}